src/useFormState.js

```diff
@@ -140,53 +175,87 @@ export default function useFormState(initialState, options) {
         if (isCheckbox || isRadio) {
           return toString(ownValue);
         }
+
+        if (isRaw) {
+          // Return `undefined` if we don't have the value.
+          return formState.current.values[name];
+        }
+
         return hasValueInState ? formState.current.values[name] : '';
       },
       onChange: callbacks.getOrSet(ON_BLUR_HANDLER + key, e => {
         setDirty(name, true);
-        let { value } = e.target;
-        if (isCheckbox) {
-          value = getNextCheckboxValue(e);
+        let value;
+        if (isRaw) {
+          value = inputOptions.onChange(e);
+          if (process.env.NODE_ENV === 'development') {
+            if (value === undefined) {
+              warn(
+                key,
+                'onChangeUndefined',
+                `You used onChange() for input ${name} with raw(), but didn't return a value!`,
+              );
+            }
+          }
+        } else {
+          if (isCheckbox) {
+            value = getNextCheckboxValue(e);
+          } else if (isSelectMultiple) {
+            value = getNextSelectMultipleValue(e);
+          } else {
+            ({ value } = e.target);
+          }
+          inputOptions.onChange(e);
         }
-        if (isSelectMultiple) {
-          value = getNextSelectMultipleValue(e);
+
+        // Mark raw fields as touched on change, since we might not get an
+        // `onBlur` event from them.
+        if (inputOptions.touchOnChange) {
+          touch(e);
         }
 
         const partialNewState = { [name]: value };
         const newValues = { ...formState.current.values, ...partialNewState };
 
         formOptions.onChange(e, formState.current.values, newValues);
-        inputOptions.onChange(e);
 
         if (!inputOptions.validateOnBlur) {
           validate(e, newValues);
         }
 
         formState.setValues(partialNewState);
       }),
-      onBlur: callbacks.getOrSet(ON_CHANGE_HANDLER + key, e => {
-        if (!formState.current.touched[name]) {
-          formState.setTouched({ [name]: true });
-          formOptions.onTouched(e);
-        }
+      onBlur: callbacks.getOrSet(
+        ON_CHANGE_HANDLER + key,
+        inputOptions.touchOnChange
+          ? undefined
```

Comment:
Hmm I'm not sure why would want to ignore onBlur completely on `touchOnChange`.

I'm thinking that we might want to keep the onBlur handler even if `touchOnChange` is set to true. Touching an input, doesn't necessarily we should ignore the blur event entirely. For example, `inputOptions.onBlur` and `formOptions.onBlur` will never be fired.

My goal with `touchOnChange` is to be used as a workaround companion to custom inputs that don't support `onBlur` – if the input doesn't support `onBlur`, it shouldn't be a problem if we pass one I think it'll be ignored. If it really causes issues, I'd argue that users should override that prop manually:

```js
<input
  {...text({
    name: 'email',
    touchOnChange: true,
    onBlur: () => console.log('I expect this to continue to work'),
  })}
/>

<CustomInput
  {...raw({
    name: 'email',
    touchOnChange: true
  })}
  // There is a good chance that CustomInput will just ignore useFormState's onBlur
  // but also it _may_ cause issues if onBlur if passed
  onBlur={undefined} // so the user can override it at call site
/>
```

Should we conditionally call `touch` instead on L233?

Reply:
You've talked me into it.  :)  That is an easy way to override onBlur.

Reply:
I think we can keep calling `touch` every time in `onBlur()`.